Mathlib/Data/Nat/Bitblast.lean

```diff
@@ -0,0 +1,223 @@
+/-
+Copyright (c) 2023 by the authors listed in the file AUTHORS and their
+institutional affiliations. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Harun Khan, Abdalrhman M Mohamed
+-/
+
+import Mathlib.Data.Nat.Bitwise
+import Mathlib.Data.Nat.ModEq
+import Mathlib.Data.Bool.Basic
+
+/-!
+# Bitblasting for Bitvectors
+
+A bitvector is defined in BitVec as a natural number modulo 2^w (i.e. `Fin` of length 2^w). 
+Many operators on bitvectors are defined as the corresponding operations on `Fin`. 
+For example, less than is defined as the less than of the underlying naturals. Similarly, addition 
+is defined as addition of the underlying naturals modulo 2^w (i.e. `Fin.add`). 
+The bitvector operations were defined from the 
+[`QF_BV` logic](https://smtlib.cs.uiowa.edu/logics-all.shtml#QF_BV) of SMT-LIBv2.
+
+In this file, we define bitblast versions for these operations and show that they correspond 
```

Comment:
Please add a reference in `references.bib`, and provide some explanation here of what `bitblast` means. Most readers will not know what this refers to.